core/src/main/java/brooklyn/entity/rebind/RebindIteration.java

```diff
@@ -341,38 +340,64 @@ protected void rebuildCatalog() {
                 }
             }
         }
+
+        // See notes in CatalogInitialization
         
-        // Register catalogue items with the management context. Loads the bundles in the OSGi framework.
-        CatalogLoadMode catalogLoadMode = managementContext.getConfig().getConfig(BrooklynServerConfig.CATALOG_LOAD_MODE);
+        Collection<CatalogItem<?, ?>> catalogItems = rebindContext.getCatalogItems();
+        CatalogInitialization catInit = ((ManagementContextInternal)managementContext).getCatalogInitialization();
+        catInit.applyCatalogLoadMode();
+        Collection<CatalogItem<?,?>> itemsForResettingCatalog = null;
+        boolean needsInitialCatalog;
         if (rebindManager.persistCatalogItemsEnabled) {
-            boolean shouldResetCatalog = catalogLoadMode == CatalogLoadMode.LOAD_PERSISTED_STATE
-                    || (!isEmpty && catalogLoadMode == CatalogLoadMode.LOAD_BROOKLYN_CATALOG_URL_IF_NO_PERSISTED_STATE);
-            boolean shouldLoadDefaultCatalog = catalogLoadMode == CatalogLoadMode.LOAD_BROOKLYN_CATALOG_URL
-                    || (isEmpty && catalogLoadMode == CatalogLoadMode.LOAD_BROOKLYN_CATALOG_URL_IF_NO_PERSISTED_STATE);
-            if (shouldResetCatalog) {
-                // Reset catalog with previously persisted state
-                logRebindingDebug("RebindManager resetting management context catalog to previously persisted state");
-                managementContext.getCatalog().reset(rebindContext.getCatalogItems());
-            } else if (shouldLoadDefaultCatalog) {
-                // Load catalogue as normal
-                // TODO in read-only mode, should do this less frequently than entities etc
-                logRebindingDebug("RebindManager loading default catalog");
-                ((BasicBrooklynCatalog) managementContext.getCatalog()).resetCatalogToContentsAtConfiguredUrl();
+            if (!catInit.hasRunOfficial() && catInit.isInitialResetRequested()) {
```

Comment:
How does catalog initialization work with the different HA states? It doesn't matter because we get a brand new ManagementContext with each rebind, which will do the full catalog init procedure, is that right?
If so should we use the rebind logger logic during catalog initialization as well to prevent excessive logging?

What happens if we go master -> standby -> master, either due to persistence store issues or force manually through rest? Will the catalog be re-initialized on each promotion?